### PR TITLE
add the ability to set the user-agent string on requests from the `Client`

### PR DESCRIPTION
**Pull request by dekkagaijin**
```release-note
NONE
```

### CODE REVIEW

File: pkg/client/client_test.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	})
+
+	testReq, err := http.NewRequest("GET", "http://www.example.com/test", nil)
+	if err != nil {
+		t.Fatalf("http.NewRequest() failed: %v", err)
+	}
+
+	testResp := &http.Response{
+		Status:     "OK",
+		StatusCode: 200,
```
Comment (mattmoor) [inline pkg/client/client_test.go:77]:
```suggestion
		StatusCode: http.StatusOK,
```

File: pkg/client/client_test.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+func TestCreateRoundTripper(t *testing.T) {
+	t.Run("always returns non-nil", func(t *testing.T) {
+		got := createRoundTripper(nil, &options{})
+		if got == nil {
+			t.Errorf("createRoundTripper() should never return a nil `http.RoundTripper`")
+		}
+	})
+
+	testReq, err := http.NewRequest("GET", "http://www.example.com/test", nil)
```
Comment (mattmoor) [inline pkg/client/client_test.go:70]:
```suggestion
	testReq, err := http.NewRequest(http.MethodGet, "http://www.example.com/test", nil)
```

Comment (dekkagaijin) [reply pkg/client/client_test.go]: done

File: pkg/client/client_test.go (hunk 1 above)
Comment (dekkagaijin) [reply pkg/client/client_test.go]: done

**Review by mattmoor — APPROVED** [review]

**Pull request merged**